arcaflow_plugin_pcp/pcp_plugin.py

```diff
@@ -144,27 +142,42 @@ def start_pcp(
         pcp2json_cmd.extend(metrics)
         print(f"Reporting metrics for: {params.pmlogger_metrics}")
 
-        try:
-            pcp_out = (
-                (
-                    subprocess.check_output(
-                        pcp2json_cmd,
-                        text=True,
-                        stderr=subprocess.STDOUT,
+        max_retries = 1
+        retries = 0
+        while retries <= max_retries:
```

Comment:
Does this still need to be a loop?

Reply:
I implemented a bit of a trick here, but I'm not positive it is the best approach. The `try` code below this does need loop control for at least 2 runs, but within the `while` loop we expect to return the function, either as `success` or as `error`, so the loop should never have a natural exit, which is the reason for the "unknown failure" capture outside of the `while` loop at line 180.

Reply:
Okay. I see. Though I do not think it's the most readable.